src/org/jetbrains/java/decompiler/modules/decompiler/SwitchHelper.java

```diff
@@ -313,14 +316,18 @@ private static boolean simplifyNewEnumSwitch(SwitchStatement switchSt) {
     if (inner instanceof InvocationExprent && ((InvocationExprent) inner).getName().equals("ordinal")) {
       InvocationExprent invInner = (InvocationExprent) inner;
 
-      ClassesProcessor.ClassNode classNode = DecompilerContext.getClassProcessor().getMapRootClasses().get(invInner.getClassname());
+      StructClass classStruct = DecompilerContext.getStructContext().getClass(invInner.getClassname());
+      if (classStruct == null) {
+        root.addComment("$VF: Unable to simplify switch-on-enum. You are probably missing an external library", true);
```

Comment:
I think it'd be best to leave the `root.addComment` line out of the PR, since it will only be written for Java 21+ class files. In addition, I think the phrasing should be different as instead of mentioning that an external library might be missing (which could be incorrect!) we should say something along the lines of: `$VF: Unable to simplify switch-on-enum, as the enum class was not able to be found.` This would be better done in a different PR though, keeping this one purely a bugfix.

Reply:
Why in different PR? This comment is relevant to this bugfix. We need to notify the user that vineflower detected this switch-on-enum form and something failed. Anyway, changed the message to your proposal.

Reply:
My point is that it should warn for the other forms too, not just the J21+ one. I suppose it's fine though; I can take a look at that one at a later time.